federation/pkg/federation-controller/service/endpoint_helper_test.go

```diff
@@ -17,11 +17,32 @@ limitations under the License.
 package service
 
 import (
+	"sync"
 	"testing"
 
+	"k8s.io/kubernetes/federation/pkg/dnsprovider/providers/google/clouddns" // Only for unit testing purposes.
 	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/util/sets"
 )
 
+var fakeDns, _ = clouddns.NewFakeInterface()
+var fakeDnsZones, _ = fakeDns.Zones()
```

Comment:
Because fake interface always supports the required interfaces.  I'll add a comment to clarify.